src/server/frontend_xwayland/xwayland_client_manager.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2020 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: William Wold <william.wold@canonical.com>
+ */
+
+#include "xwayland_client_manager.h"
+#include "xwayland_log.h"
+#include "null_event_sink.h"
+#include "mir/shell/shell.h"
+#include "mir/scene/session.h"
+#include "mir/log.h"
+
+#include "boost/throw_exception.hpp"
+
+namespace mf = mir::frontend;
+namespace msh = mir::shell;
+namespace ms = mir::scene;
+
+mf::XWaylandClientManager::XWaylandClientManager(std::shared_ptr<shell::Shell> const& shell)
+    : shell{shell}
+{
+}
+
+mf::XWaylandClientManager::~XWaylandClientManager()
+{
+    // release() should have been called for every owner before destroying
+    if (!sessions_by_pid.empty() || !sessions_by_owner.empty())
+    {
+        log_warning(
+            "XWaylandClientManager destroyed without all sessions being released "
+            "(sessions_by_pid: %lu, sessions_by_owner: %lu)",
+            sessions_by_pid.size(),
+            sessions_by_owner.size());
```

Comment:
As the compiler mentions, sizes should be formatted with %z.

Reply:
Fixed